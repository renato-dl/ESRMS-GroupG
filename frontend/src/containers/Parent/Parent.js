import React from 'react';
import { Icon} from 'semantic-ui-react';
import { api } from '../../services/api';
import { StudentCard } from '../../components/StudentCard/StudentCard';
import './Parent.scss';

export class Parent extends React.Component {
  state = {
    children: [
  }

  async componentDidMount() {
    const response = await api.parent.getChilds('9d64fa59c91d9109b11cd9e05162c675');
    if (response) {
      this.setState({children: response.data})
    }
  }

  selectChild = async (childId) => {
    console.log(childId);

    this.props.history.push(`/parent/student/${childId}`);
  }

  render() {
    return (
      <div className="parent-container">
        <h3 className="contentHeader">
        <Icon name='braille' size="small" />
        Select/Switch child</h3>
        {/* <h2 className="title">My children</h2> */}
        <div className="children">
          {this.state.children.map((child, index) => (
            <StudentCard 
              key={index}
              {...child} 
              selected={child.id === this.state.selectedChild}
              onClick={() => this.selectChild(child.ID)} 
            />
          ))}
        </div>
      </div>
    )
  }
}
