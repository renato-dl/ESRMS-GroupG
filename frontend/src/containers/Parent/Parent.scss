.parent-container {
  width: calc(100% - 260px);
  .title {
    text-align: center;
  }
  
  .children {
    display: flex;
    align-items: center;
    justify-content: center;
  }
}