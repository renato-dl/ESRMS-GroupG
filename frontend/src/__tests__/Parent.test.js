import React from 'react';
import {Parent} from '../containers/Parent/Parent';
import {getRouterPropsForTest} from '../utils';
import {shallow} from 'enzyme';

describe('Testing Parent component', () => {
  
  test('Test if Parent component is rendered', () => {
    shallow(
      <Parent {...getRouterPropsForTest()} />
    );
  });

});
