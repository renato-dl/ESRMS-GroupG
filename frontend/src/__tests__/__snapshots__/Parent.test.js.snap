// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Parent component Testing snapshot 1`] = `
<div
  className="parent-container"
>
  <h2
    className="title"
  >
    My children
  </h2>
  <div
    className="children"
  />
</div>
`;
